package.json

```diff
@@ -370,7 +370,23 @@
         "category": "Ruff",
         "command": "ruff.restart"
       }
-    ]
+    ],
+    "menus": {
+      "commandPalette": [
+        {
+          "command": "ruff.executeAutofix",
+          "when": "resourceExtname == .py || resourceExtname == .pyi || resourceExtname == .ipynb"
+        },
+        {
+          "command": "ruff.executeFormat",
+          "when": "resourceExtname == .py || resourceExtname == .pyi || resourceExtname == .ipynb"
+        },
+        {
+          "command": "ruff.executeOrganizeImports",
+          "when": "resourceExtname == .py || resourceExtname == .pyi || resourceExtname == .ipynb"
```

Comment:
Hmm, I don't think we can do that. You can configure ruff to lint files with arbitrary extensions using `extend-include`. These files would no longer be linted. It also means that auto fix would no longer run on any new/untitled files because they don't have an extension yet (they only have an associated language that we should probably use in the LSP).

Reply:
You're right, this wouldn't work with arbitrary file extensions.

I might try using @dhruvmanila's suggestion and just handle the `snapshot not available` error more gracefully in `ruff server`.